### PR TITLE
Implemented forum bridge

### PR DESCRIPTION
**Pull request by samdark**
#218

### CODE REVIEW

File: components/ForumAdapter.php [hunk 1]
```diff
@@ -2,44 +2,196 @@
...
+                'language' => 1,
+
+                'members_pass_hash' => $this->getIPBPasswordHash($ipbSalt, $password),
+                'members_pass_salt' => $ipbSalt,
+            ])->execute();
+
+            $forumUserId = $this->db->getLastInsertID();
+        }
+
+        $user->id = $forumUserId;
```
Comment (cebe) [inline components/ForumAdapter.php]: overwriting user pk does not make sense if user model is already saved, like its the case in AuthHandler.php above. its better to store ipb ID in a separate column.

File: components/ForumAdapter.php [hunk 2]
```diff
@@ -2,44 +2,196 @@
...
+     * @param User $user
+     * @param string $password
+     */
+    public function createForumUser(User $user, $password)
+    {
+        if (!$user->isNewRecord) {
+            throw new InvalidArgumentException('User should be new.');
+        }
+
+        $ipbSalt = $this->generateIPBPasswordSalt();
```
Comment (cebe) [inline components/ForumAdapter.php]: can be moved inside the `if (!$forumUserId)` block

File: components/ForumAdapter.php (hunk 1 above)
Comment (cebe) [reply components/ForumAdapter.php]: even if these IDs may be the same on many records, having a separate column allows us more flexibility in the future.

**Comment by samdark** [comment]
@cebe adjusted.

**Pull request merged**